File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.cpp
class Solution {
public:
    int minimumPushes(string word) {
        vector<int> freq(26,0);
        for(auto &w : word){
            freq[w - 'a']++;
        }
        sort(freq.rbegin(),freq.rend());
        int it = 1, sum = 0;
        for(int i = 0 ; i < 26 ; i++){
            sum += it * freq[i];
            if((i + 1) % 8 == 0) it++;
        }
        return sum ;
    }
};

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.go
func minimumPushes(word string) int {
    freq := make([]int, 26)
    for _, w := range word {
        freq[w-'a']++
    }

    sort.Sort(sort.Reverse(sort.IntSlice(freq)))

    it := 1
    sum := 0
    for i := 0; i < 26; i++ {
        sum += it * freq[i]
        if (i+1)%8 == 0 {
            it++
        }
    }

    return sum
}
